core/src/ids.rs

```diff
@@ -98,17 +98,26 @@ macro_rules! declare_id {
 
         impl core::fmt::Display for $name {
             fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-                let mut end = self.0.len();
+                let len = self.0.len();
+                let mut e1 = len / 2;
+                let mut s2 = len / 2;
```

Comment:
It's the case but this division is not good for odd `len`. As an option:

```rust
let mut e1 = len / 2;
let mut s2 = len - e1;
```

Reply:
why?
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d3e7b4350ef3b42649f436adaac4b253)
`let s = len - e1` will skip one element 

But that's really not our case

Reply:
Your solution is the best!